models/src/main/scala/coop/rchain/models/rholangn/parmanager/SerializedSize.scala

```diff
@@ -0,0 +1,140 @@
+package coop.rchain.models.rholangn.parmanager
+
+import com.google.protobuf.CodedOutputStream
+import coop.rchain.models.rholangn._
+
+import scala.annotation.unused
+
+private[parmanager] object SerializedSize {
+
+  import Constants._
+
+  private def sSize(bytes: Array[Byte]): Int = CodedOutputStream.computeByteArraySizeNoTag(bytes)
+
+  private def sSize(@unused v: Boolean): Int = booleanSize
+  private def sSize(v: Int): Int             = CodedOutputStream.computeInt32SizeNoTag(v)
+  private def sSize(v: Long): Int            = CodedOutputStream.computeInt64SizeNoTag(v)
+  private def sSize(v: BigInt): Int          = sSize(v.toByteArray)
+  private def sSize(v: String): Int          = CodedOutputStream.computeStringSizeNoTag(v)
+
+  private def sSize(p: RhoTypeN): Int              = p.serializedSize
+  private def sSize(kv: (RhoTypeN, RhoTypeN)): Int = kv._1.serializedSize + kv._2.serializedSize
+  private def sSizeInjection(injection: (String, RhoTypeN)): Int =
+    sSize(injection._1) + injection._2.serializedSize
+
+  private def sSizeSeq[T](seq: Seq[T], f: T => Int): Int =
+    sSize(seq.size) + seq.map(f).sum
+
+  private def sSize(ps: Seq[RhoTypeN]): Int = sSizeSeq[RhoTypeN](ps, sSize)
+
+  private def sSizeStrings(strings: Seq[String]): Int = sSizeSeq[String](strings, sSize)
+
+  private def sSizeKVPairs(strings: Seq[(RhoTypeN, RhoTypeN)]): Int =
+    sSizeSeq[(RhoTypeN, RhoTypeN)](strings, sSize)
+
+  private def sSizeInjections(injections: Seq[(String, RhoTypeN)]): Int =
+    sSizeSeq[(String, RhoTypeN)](injections, sSizeInjection)
+
+  private def sSize(pOpt: Option[RhoTypeN]): Int =
+    booleanSize + (if (pOpt.isDefined) pOpt.get.serializedSize else 0)
```

Comment:
When you have an Option you can map into it and adjust the value.

```scala
booleanSize  + pOpt.map(_.serializedSize).getOrElse(0)
```
or I think better
```scala
pOpt.map(_.serializedSize + booleanSize).getOrElse(booleanSize)
```

Reply:
This is how to serialized option value:
```scala
      private def write(pOpt: Option[RhoTypeN]): Unit =
        if (pOpt.isDefined) {
          write(true)
          write(pOpt.get)
        } else write(false)
```

Reply:
I see, so I is not possible to just write nothing?

Reply:
Yes, we need to save minimum 1 byte, this will show `None`.